### PR TITLE
replace stringy mols with lists

### PR DESCRIPTION
**Pull request by blazejba**
replaces stringy mols with lists in:
- `nanoDFT/sparse_ERI.py`
- `nanoDFT/symmetric_ERI.py`
- `density_functional_theory.py`
- `nanoDFT/experimental_pmap_nanoDFT.py`
- `nanoDFT/nanoDFT.py`

### CODE REVIEW

File: density_functional_theory.py [hunk 1]
```diff
@@ -1411,30 +1411,37 @@ def jax_dft(str):
             recompute(args, None, 0, 0, our_fun=jax_dft, str=string)
```
Comment (awf) [inline density_functional_theory.py:1411]: I wonder about also generally replacing `mol_str` with `mol`, and renaming the `str=` argument to `recompute` to `mol=`?  But that can be another PR if you prefer.

**Review by awf — APPROVED** [review]
Ahh, yes, much nicer :)

File: density_functional_theory.py [hunk 2]
```diff
@@ -1411,30 +1411,37 @@ def jax_dft(str):
...
+            ["C", ( 1.28780, -1.45030,  0.16290)],
+            ["C", ( 1.28960, -0.65950,  1.30470)],
+            ["C", ( 0.01150, -0.64600,  1.85330)],
+            ["C", ( 1.58300,  0.64540,  0.89840)],
+            ["C", ( 0.48480,  1.43830,  1.19370)],
+            ["C", (-0.50320,  0.64690,  1.77530)],
+            ["C", (-1.60620,  0.67150,  0.92310)],
+            ["C", (-1.29590,  1.48910, -0.16550)],
+            ["C", (-0.01020,  1.97270, -0.00630)]
+        ][:args.C]
```
[the remark this answers is not in the document]
Comment (blazejba) [reply density_functional_theory.py:1435]: `python density_functional_theory.py -C 8 -backend cpu` works, although `| energies - pyscf_energies |` is quite large:
```
> diffs:
[0.00878442 0.00747065 0.00654506 0.00875562 0.00908959]
mean+-var: 0.008129 +- 0.000001
```

To test a PR locally you can always do `git checkout less_stringy_mols` or whatever the name of the branch is

File: density_functional_theory.py (hunk 1 above)
Comment (blazejba) [reply density_functional_theory.py:1411]: `mol` is already used for the objects of `pyscf.gto.mole.Mole` class; we need a different name :)
